src/js/api.ts

```diff
@@ -557,12 +557,28 @@ export class PyodideAPI {
     );
 
     // sync native ==> browser
-    await new Promise((resolve, _) => Module.FS.syncfs(true, resolve));
+    await new Promise<void>((resolve, reject) => {
+      Module.FS.syncfs(true, (err: any) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve();
+        }
+      });
+    });
 
     return {
       // sync browser ==> native
       syncfs: async () =>
-        new Promise((resolve, _) => Module.FS.syncfs(false, resolve)),
+        new Promise((resolve, reject) =>
```

Comment:
not sure how strict you folks are around *TS* but `<void>` is missing in this sync `false` case. True that it wasn't there in the first place and *TS* can infer that so I let you decide if you should make it consistent here or before, removing the other `<void>`.